"Rename Symbol" does not report language server errors

### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

Trying "Rename Symbol" with a macro does not work.

To reproduce:
1. Select a macro
2. Right click and select "Rename Symbol" or use the shortcut (F2)

Actual result:
Nothing happens

Expected result:
Show the rename editable text input

Looking at the log, we see that this operation is not supported

```
2025-01-10T19:31:24.457366+02:00 [ERROR] Cannot rename symbol: symbol is not a supported kind (e.g. namespace, macro)
```

But there is no indication that the operation is not supported.

I think we have several issues:
- Not all kind of symbols are supported
- The "Rename Symbol" menu item is enabled while the current select text is not a supported kind
- There is no indication of the error - logging is not good enough, even when we have an easy way to open the log

### Zed Version and System Specs

Zed: v0.163.3 (Zed)
OS: macOS 15.2.0
Memory: 32 GiB
Architecture: aarch64

### If applicable, add screenshots or screencasts of the incorrect state / behavior

_No response_

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary>

<!-- Click below this line and paste or drag-and-drop your log-->
```
2025-01-10T19:31:24.456762+02:00 [WARN] Generic lsp request to clangd failed: Cannot rename symbol: symbol is not a supported kind (e.g. namespace, macro)
2025-01-10T19:31:24.457366+02:00 [ERROR] Cannot rename symbol: symbol is not a supported kind (e.g. namespace, macro)
```
<!-- Click above this line and paste or drag-and-drop your log--></details>

--- a/crates/editor/src/element.rs
+++ b/crates/editor/src/element.rs
@@ -73,7 +73,7 @@ use ui::{
 };
 use unicode_segmentation::UnicodeSegmentation;
 use util::{RangeExt, ResultExt};
-use workspace::{item::Item, Workspace};
+use workspace::{item::Item, notifications::NotifyTaskExt, Workspace};
 
 struct SelectionLayout {
     head: DisplayPoint,
@@ -314,32 +314,38 @@ impl EditorElement {
         register_action(view, cx, Editor::go_to_next_hunk);
         register_action(view, cx, Editor::go_to_prev_hunk);
         register_action(view, cx, |editor, a, cx| {
-            editor.go_to_definition(a, cx).detach_and_log_err(cx);
+            editor.go_to_definition(a, cx).detach_and_notify_err(cx);
         });
         register_action(view, cx, |editor, a, cx| {
-            editor.go_to_definition_split(a, cx).detach_and_log_err(cx);
+            editor
+                .go_to_definition_split(a, cx)
+                .detach_and_notify_err(cx);
         });
         register_action(view, cx, |editor, a, cx| {
-            editor.go_to_declaration(a, cx).detach_and_log_err(cx);
+            editor.go_to_declaration(a, cx).detach_and_notify_err(cx);
         });
         register_action(view, cx, |editor, a, cx| {
-            editor.go_to_declaration_split(a, cx).detach_and_log_err(cx);
+            editor
+                .go_to_declaration_split(a, cx)
+                .detach_and_notify_err(cx);
         });
         register_action(view, cx, |editor, a, cx| {
-            editor.go_to_implementation(a, cx).detach_and_log_err(cx);
+            editor.go_to_implementation(a, cx).detach_and_notify_err(cx);
         });
         register_action(view, cx, |editor, a, cx| {
             editor
                 .go_to_implementation_split(a, cx)
-                .detach_and_log_err(cx);
+                .detach_and_notify_err(cx);
         });
         register_action(view, cx, |editor, a, cx| {
-            editor.go_to_type_definition(a, cx).detach_and_log_err(cx);
+            editor
+                .go_to_type_definition(a, cx)
+                .detach_and_notify_err(cx);
         });
         register_action(view, cx, |editor, a, cx| {
             editor
                 .go_to_type_definition_split(a, cx)
-                .detach_and_log_err(cx);
+                .detach_and_notify_err(cx);
         });
         register_action(view, cx, Editor::open_url);
         register_action(view, cx, Editor::open_selected_filename);
@@ -382,14 +388,14 @@ impl EditorElement {
         register_action(view, cx, Editor::expand_all_hunk_diffs);
         register_action(view, cx, |editor, action, cx| {
             if let Some(task) = editor.format(action, cx) {
-                task.detach_and_log_err(cx);
+                task.detach_and_notify_err(cx);
             } else {
                 cx.propagate();
             }
         });
         register_action(view, cx, |editor, action, cx| {
             if let Some(task) = editor.format_selections(action, cx) {
-                task.detach_and_log_err(cx);
+                task.detach_and_notify_err(cx);
             } else {
                 cx.propagate();
             }
@@ -399,42 +405,42 @@ impl EditorElement {
         register_action(view, cx, Editor::show_character_palette);
         register_action(view, cx, |editor, action, cx| {
             if let Some(task) = editor.confirm_completion(action, cx) {
-                task.detach_and_log_err(cx);
+                task.detach_and_notify_err(cx);
             } else {
                 cx.propagate();
             }
         });
         register_action(view, cx, |editor, action, cx| {
             if let Some(task) = editor.compose_completion(action, cx) {
-                task.detach_and_log_err(cx);
+                task.detach_and_notify_err(cx);
             } else {
                 cx.propagate();
             }
         });
         register_action(view, cx, |editor, action, cx| {
             if let Some(task) = editor.confirm_code_action(action, cx) {
-                task.detach_and_log_err(cx);
+                task.detach_and_notify_err(cx);
             } else {
                 cx.propagate();
             }
         });
         register_action(view, cx, |editor, action, cx| {
             if let Some(task) = editor.rename(action, cx) {
-                task.detach_and_log_err(cx);
+                task.detach_and_notify_err(cx);
             } else {
                 cx.propagate();
             }
         });
         register_action(view, cx, |editor, action, cx| {
             if let Some(task) = editor.confirm_rename(action, cx) {
-                task.detach_and_log_err(cx);
+                task.detach_and_notify_err(cx);
             } else {
                 cx.propagate();
             }
         });
         register_action(view, cx, |editor, action, cx| {
             if let Some(task) = editor.find_all_references(action, cx) {
-                task.detach_and_log_err(cx);
+                task.detach_and_notify_err(cx);
             } else {
                 cx.propagate();
             }

